statuscompliance/status-backend
# Feature/27 store node red computations

#### Description
This PR introduces the design and documentation for the Computation endpoints in the API. The following changes have been made:

- **Endpoint Design**: Developed RESTful endpoints for managing computations, including:
  - Retrieving all computations
  - Retrieving a single computation by ID
  - Retrieving computations by control ID
  - Creating a new computation
  - Creating multiple computations in bulk
  - Deleting all computations
  - Deleting computations by control ID

- **Swagger Documentation**: Comprehensive documentation has been added for the newly designed endpoints, including:
  - Descriptions of each endpoint
  - Parameters and response schemas
  - Status codes and error handling

#### Changes Made
- Implemented the following endpoints:
  - `GET /api/computation`
  - `GET /api/computation/{id}`
  - `GET /api/control/{control_id}/computations`
  - `POST /api/computation`
  - `POST /api/computations`
  - `DELETE /api/computation`
  - `DELETE /api/control/{control_id}/computations`
  
- Added Swagger annotations for the endpoints for better API visibility and usability.

#### How to Test
1. Pull this branch and run the API.
2. Use a tool like Postman or Swagger UI to test the endpoints.
3. Verify that the responses match the documented schemas.
